src/pycrunch/importing.py

```diff
@@ -52,6 +52,40 @@ def wait_for_batch_status(self, batch, status):
         else:
             raise ValueError("The batch did not reach the '%s' state in the "
                              "given time. Please check again later." % status)
+    
+    def add_schema_metadata(self, site, schema, metadata, filename, fp, mimetype="application/x-parquet"):
+        """
+        Create a new Source from a parquet file using schema and metadata.
+
+        Parameters:
+        site (shoji.Catalog): a shoji Catalog object, from which we acquire session and sources url
+        schema (str): json string containing schema
+        metadata (str): json string containing metadata
+        filename (str): name of file being uploaded
+        fp (BufferedReader): opened file object
+        mimetype (str): mimetype of file being uploaded
+
+        Returns:
+        shoji.Entity: Shoji entity containing the payload, status_code and source_url of uploaded file
+        """
+        response = site.session.post(
+            site.catalogs.sources, 
+            files={
+                "uploaded_file": (filename, fp, mimetype)
+            },
+            data={
+                "schema": schema, 
+                "metadata": metadata,
+                "crunchlake": "create",
```

Comment:
What is this argument?

Reply:
It's crunchlake specific. @mgdotdev can expand on this.